### PR TITLE
New interfaces: safety check functions

### PR DESCRIPTION
**Pull request by michael-smirnov**

### CODE REVIEW

File: cpp/oneapi/dal/util/detail/common.hpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+template <typename Data>
+inline void throw_on_sum_overflow(const Data& first, const Data& second) {
+    static_assert(std::is_integral_v<Data>, "The check requires integral operands");
+
+    volatile Data tmp = first + second;
+    tmp -= first;
+    if (tmp != second) {
+        throw range_error("overflow found in sum of two values");
+    }
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/util/detail/common.hpp:50]: Why don't you use `throw_on_false`?

 ```suggestion
    throw_on_false<range_error>(tmp == second, "overflow found in sum of two values");
```

File: cpp/oneapi/dal/util/detail/common.hpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+template <typename Data>
+inline void throw_on_mul_overflow(const Data& first, const Data& second) {
+    static_assert(std::is_integral_v<Data>, "The check requires integral operands");
+
+    if (first != 0 && second != 0) {
+        volatile Data tmp = first * second;
+        tmp /= first;
+        if (tmp != second) {
+            throw range_error("overflow found in multiplication of two values");
+        }
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/util/detail/common.hpp:62]:
```suggestion
        throw_on_false<range_error>(tmp == second, "overflow found in multiplication of two values");
```

**Review by KulikovNikita — APPROVED** [review]

**Comment by michael-smirnov** [comment]
File with checks: https://github.com/oneapi-src/oneDAL/pull/1137/files#diff-d0e19bf33dd8eca1c4cae781adcdde2747203ff72b258076f15a25e2a4746eb2

**Review by rlnx — APPROVED** [review]

**Pull request merged**